### ISSUE TITLE
Message-utils defaults to sender when recipient is explicit

### ISSUE THREAD

**Issue by dianabarsan** [medic/medic#5369]
This is how the recipient phone number is determined at the moment:
https://github.com/medic/medic/blob/acff94a24775080ffda5927570757c6c0ffe99b6/shared-libs/message-utils/src/index.js#L105

where 
- `phone` is valid if `recipient` is an expression (`clinic`, `reporting_unit`, `report.contact.phone` etc) that is valid against the context
- `from` is `report.from || (report.contact && report.contact.phone)`
- `recipient` is the explicit recipient (which could be an actual phone number)

If you'd want to generate a message that should be sent to a specific explicit phone number, you will only succeed if the generating report does not have a `from` or `contact.phone`.

**TLDR**
Transitions configured with explicit recipients and `multi_report_alerts` will not create messages with correct recipients.

**Steps to reproduce**:
- Pick a simple transition that can add a message with a specific recipient, for example: `conditional_alerts`. Example will continue with specific scenario configuring `conditional_alerts`.
- Enable the transition
- Add the following configuration (or similar):
```
"alerts": [{
  "form": "P",
  "condition": "true",
  "message": "this is my test message",
  "recipient": "+40755203040"
}]
```
(you will likely already have an `alerts` property in your settings, either replace or add as a new entry).
- Go to admin and submit a P form (from a different number that the one configured as your alert recipient, it can even be an unknown phone number!)
- Allow sentinel to process the report and check it

**What should happen**:
- The report should have a task with "this is my test message" message to be sent to "+40755203040"

**What actually happens**:
- The report has a task with "this is my test message" message to be sent to the sender phone number.

As an alternative scenario, and the way I discovered this error, you can use `multi_report_alerts` transition.
Add a configuration like:

```
 "multi_report_alerts": [{
  "name": "test",
  "is_report_counted": "function(r, l) { return true }",
  "num_reports_threshold": 2,
  "message": "multi_report_message",
  "recipients": ["new_report.from"],
  "time_window_in_days": 1,
  "forms": ["N"]
}],
```

Then submit 2 N forms from different phone numbers. 
What you would expect is that your second form should have 2 tasks with `multi_report_message` message and to be sent to the 2 numbers that you submitted N forms with.
Instead, both tasks are marked to be sent to the last submitter's phone, with one of the tasks marked with `duplicate` state.

This effectively breaks `multi_report_alerts` transition, as it parses the phones before generating messages and provides explicit phone numbers to `message-utils`, thus all messages will be sent to the last submitter's phone instead of the configured recipients.

**Comment by garethbowen** [comment]
Nice one!

I've merged it into `master` - I'll leave it with you to publish to npm and update dependencies.

**Comment by garethbowen** [comment]
AT passed.

- `"123"` -> 123
- `""` -> sender's phone number
- `"clinic"` -> the clinic's contact phone number
